rotkehlchen/utils/snapshots.py

```diff
@@ -0,0 +1,98 @@
+        tuple(location_data[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
+    )
+    if has_invalid_headers:
+        return False, 'csv file has invalid headers'
+
+    # check if the timestamp can be converted to int
+    try:
+        balances_timestamps = [deserialize_timestamp(entry['timestamp']) for entry in balances_data]  # noqa: E501
+        location_data_timestamps = [deserialize_timestamp(entry['timestamp']) for entry in location_data]  # noqa: E501
+    except DeserializationError:
+        return False, 'csv file contains invalid timestamp format'
+    # check if all timestamps are the same.
+    has_different_timestamps = (
+        balances_timestamps.count(balances_timestamps[0]) != len(balances_timestamps) or
+        location_data_timestamps.count(location_data_timestamps[0]) != len(location_data_timestamps) or  # noqa: E501
+        balances_timestamps[0] != location_data_timestamps[0]
+    )
+    if has_different_timestamps is True:
+        return False, 'csv file has different timestamps'
+
+    return True, ''
+
+
+def parse_import_snapshot_data(
+        balances_snapshot_file: Path,
+        location_data_snapshot_file: Path,
+) -> Tuple[bool, Dict[str, Any]]:
+    """This function takes a path to the snapshot and converts it to a list of dictionaries.
+    It runs a series of validation checks, returns the status of the validation
+    and a dictionary containing an error message and the converted snapshot data.
+    """
+    balances_list = _csv_to_dict(balances_snapshot_file)
+    location_data_list = _csv_to_dict(location_data_snapshot_file)
+    is_valid, message = validate_import_data(
+        balances_data=balances_list,
+        location_data=location_data_list,
+    )
+    if is_valid is False:
+        return False, {'msg': message, 'data': {}}
+
+    return True, {
+        'msg': '',
+        'data': {'balances_list': balances_list, 'location_data_list': location_data_list},
+    }
+
+
+def _csv_to_dict(file: Path) -> List[Dict[str, str]]:
+    """Converts a csv file to a list of dictionary."""
+    with open(file, mode='r') as csv_file:
+        csv_reader = DictReader(csv_file)
+        return list(csv_reader)
+
+
+def get_main_currency_price(
+        db: DBHandler,
+        timestamp: Timestamp,
+        msg_aggregator: MessagesAggregator,
+) -> Tuple[Asset, Price]:
+    """Gets the main currency and its equivalent price at a particular timestamp."""
+    main_currency = db.get_main_currency()
+    main_currency_price = None
+    try:
+        main_currency_price = PriceHistorian.query_historical_price(
+            from_asset=A_USD,
+            to_asset=main_currency,
+            timestamp=timestamp,
+        )
+    except NoPriceForGivenTimestamp:
+        main_currency_price = Price(ONE)
+        msg_aggregator.add_error(
+            f'Could not find price for timestamp {timestamp}. Using USD for export. '
```

Comment:
mention the price for WHICH asset.
```suggestion
            f'Could not find {main_currency.symbol} price for timestamp {timestamp}. Using USD for export. '
```